src/metabase/api/search.clj

```diff
@@ -295,6 +296,12 @@
   [{:keys [id]}]
   (-> id Segment mi/can-read?))
 
+(defn- hydrate-if-necessary
```

Comment:
the name here is pretty ambiguous or general but its a specific usecase here. I'd probably do something that indicated what its doing `include-dashboard-count` or `hydrate-dashboard-count` or something similar. Another though, is that hydration makes a lot of sense for a single item. If there are 500 cards in the search results this will make 500 queries which would be terrible. For a collection, it might be necessary to get all card-ids, issue a query getting those counts, and then annotate the rows with the count. Classic N+1 query situation.

Reply:
That's a really good point about the N+1 querying. As far as I can figure I can't just pluck out all the card IDs at the end since the dashboard count is needed for the transduction.

I think the answer is to modify the original query to have a `SELECT count(*) as dashboard_count from dashboard_card where card_id = card.id`. There's some trickery with the how select clause is made, though, which I'll need to explore in more depth.